flink-runtime/src/main/java/org/apache/flink/runtime/deployment/CachedShuffleDescriptors.java

```diff
@@ -54,10 +56,18 @@ public class CachedShuffleDescriptors {
     /** Stores the mapping of resultPartitionId to index subscripts in consumed partition group. */
     private final Map<IntermediateResultPartitionID, Integer> resultPartitionIdToIndex;
 
-    public CachedShuffleDescriptors(
+    /** The number of consumers for {@link ConsumedPartitionGroup}. */
+    private final int numConsumers;
+
+    private final int offloadShuffleDescriptorsThreshold;
```

Comment:
Since all the values of `offloadShuffleDescriptorsThreshold` are the same, could we put this value into `TaskDeploymentDescriptorFactory`? Also, we could move the `shouldOffload` function to the `TaskDeploymentDescriptorFactory`, too. Then we just need to pass all necessary parameters to `shouldOffload` in `serializeAndTryOffloadShuffleDescriptor` function.